libr/cons/hud.c

```diff
@@ -19,6 +19,10 @@ R_API char *r_cons_hud_file(const char *f) {
 // Display a buffer in the hud (splitting it line-by-line and ignoring
 // the lines starting with # )
 R_API char *r_cons_hud_string(const char *s) {
+	if (!r_cons_is_interactive ()) {
+		eprintf ("Hud mode requires scr.interactive=true.\n");
```

Comment:
i think its better to just return NULL here. and show this eprintf in the callers from core. because scr.interactive is core-specific, not related to RCons

Reply:
I understand but both functions `r_cons_hud_string` and `r_cons_less_str` are called in multiple places (some of them not directly from the core):

```
$ grep -rn 'r_cons_hud_string (' 
libr/cons/less.c:85:			r_cons_hud_string (ostr);
libr/cons/grep.c:532:				r_cons_hud_string (cons->context->buffer);
libr/cons/grep.c:545:			char *res = r_cons_hud_string (buf);
libr/cons/more.c:81:			r_cons_hud_string (ostr);
libr/cons/hud.c:12:		char *ret = r_cons_hud_string (s);
```
```
$ grep -rn 'r_cons_less_str ('   
libr/cons/less.c:90:				(void)r_cons_less_str (r_cons_less_help, NULL);
libr/cons/less.c:169:	(void)r_cons_less_str (r_cons_singleton ()->context->buffer, NULL);
libr/cons/grep.c:535:				r_cons_less_str (cons->context->buffer, NULL);
libr/cons/grep.c:551:			r_cons_less_str (buf, NULL);
libr/cons/cons.c:903:				r_cons_less_str (str, NULL);
libr/core/cmd_log.c:233:					r_cons_less_str (b, NULL);
libr/core/visual.c:587:		ret = r_cons_less_str (r_strbuf_get (p), "?");
libr/core/visual.c:602:		ret = r_cons_less_str (r_strbuf_get (p), "?");
libr/core/visual.c:610:		ret = r_cons_less_str (r_strbuf_get (p), "?");
libr/core/visual.c:618:		ret = r_cons_less_str (r_strbuf_get (p), "?");
libr/core/visual.c:633:		ret = r_cons_less_str (r_strbuf_get (p), "?");
libr/core/visual.c:644:		ret = r_cons_less_str (r_strbuf_get (p), "?");
libr/core/visual.c:658:		ret = r_cons_less_str (r_strbuf_get (p), "?");
libr/core/visual.c:671:		ret = r_cons_less_str (r_strbuf_get (p), "?");
libr/core/visual.c:2088:		r_cons_less_str ("\nVisual Insert Mode:\n\n"
```

I think it would make the rest of the code messy

Apart from that both functions return `NULL` whether it succeeded or not and the return value is ignore in most if not all cases

Reply:
as long as the erroronly matters from core, i would say to make a wrapper of this function inside visual.c, just to show the error in case the function returns false. For the rest maybe its fine to just return silently. what do you think?